### ISSUE TITLE
(2.1) Crash with grad rectangle when using "shift" to draw "under" the menu

### ISSUE THREAD

**Issue by GoogleCodeExporter** [birdal46/grafx2#290]
```
What steps will reproduce the problem?
1. Select grad rect tool
2. Put the mouse cursor over the menu
3. Press and hold shift
4. Move the mouse a few pixels right or left
5. Move the mouse into the drawing area (The crosshair cursor will only
show a vertical line, because the horizontal position is under the image,
below the menu)
6. Start drawing a grad rectangle
7. sigsegv

It's the XOR drawing that crashes. Substraction creates negative
coordinate, unfortunately a lot of old code uses unsigned numbers and this
tries to draw at coordinate 65000+ of the screen.

Happens on both 2.1 and 2.2
```

Original issue reported on code.google.com by `yrizoud` on 16 Jan 2010 at 9:28

**Comment by GoogleCodeExporter** [comment]
```
Fixed in r1265.
```

Original comment by `pulkoma...@gmail.com` on 16 Jan 2010 at 9:59
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Oh thanks, you didn't have to do it so fast :)
I was writing it down so I wouldn't forget, while I'm in the middle of something
unrelated :The drop-down menu to show/hide toolbars.
```

Original comment by `yrizoud` on 16 Jan 2010 at 10:20

**Comment by GoogleCodeExporter** [comment]
```
Well, it's my messy code in there, so I could fix it :)
I'll let you choose if it's easy enough to trigger to migrate the fix to 2.1...
```

Original comment by `pulkoma...@gmail.com` on 16 Jan 2010 at 10:25

**Comment by GoogleCodeExporter** [comment]
```
Not worth it, it's rather elaborate, and involves a tool that isn't used so 
much.
Anyway "make ziprelease" is still broken on 2.1 branch, so I can't easily (ie:
reliably) re-package.

```

Original comment by `yrizoud` on 16 Jan 2010 at 11:29
